src/index.js

```diff
@@ -46,29 +47,53 @@ class GoogleLogin extends Component {
       });
     });
   }
-  signIn(e) {
+  maybeSetPrompt = (params) => {
+    const { prompt } = this.props;
+    // only set prompt if it was passed in
+    if (prompt) {
+      params.prompt = prompt;
+    }
+  }
+
+  maybeSetRedirectUri = (params) => {
+    const { uxMode, redirectUri } = this.props;
+    // only set redirect_uri if we're actually redirecting
+    if (uxMode === 'redirect') {
```

Comment:
Note that if no `redirect_uri` param is given, it will default to the current URL (which is generally the case). Maybe you should first check that `redirect_uri` is not null/empty? (for clarity)